virtualmachine/vsphere/vm.go

```diff
@@ -1140,7 +1140,7 @@ func DeleteTemplate(vm *VM) error {
 }
 
 // GetDatastores : Returns the datastores in a host/cluster in a cluster
-func GetDatastores(vm *VM) ([]Datastore, error) {
+func (vm *VM) GetDatastores() ([]Datastore, error) {
```

Comment:
This will lead to changes in VirtualMachine Interface and implementation for other cloud as well.